src/views/sources/viewSourcesList.tsx

```diff
@@ -421,25 +400,33 @@ const SourcesListView: React.FunctionComponent = () => {
         </ConditionalTableBody>
       </Table>
       <Pagination variant="bottom" widgetId="server-paginated-example-pagination" />
-      {!!credentialsSelected.length && (
-        <Modal
-          variant={ModalVariant.small}
-          title={t('view.label', { context: 'credentials' })}
-          isOpen={!!credentialsSelected}
-          onClose={() => setCredentialsSelected([])}
-          actions={[
-            <Button key="cancel" variant="secondary" onClick={() => setCredentialsSelected([])}>
-              Close
-            </Button>
-          ]}
-        >
-          <List isPlain isBordered>
-            {credentialsSelected.map(c => (
-              <ListItem key={c.name}>{c.name}</ListItem>
-            ))}
-          </List>
-        </Modal>
-      )}
+      <Modal
+        variant={ModalVariant.small}
+        title={t('view.label', { context: 'credentials' })}
+        isOpen={credentialsSelected.length > 0}
+        onClose={() => setCredentialsSelected([])}
+        actions={[
+          <Button
+            key="confirm"
+            variant="danger"
+            onClick={() => {
+              // Add your specific action if needed
+              setCredentialsSelected([]);
+            }}
+          >
+            {t('table.label', { context: 'close' })}
+          </Button>,
+          <Button key="cancel" variant="link" onClick={() => setCredentialsSelected([])}>
+            {t('form-dialog.label', { context: 'cancel' })}
+          </Button>
+        ]}
+      >
+        <List isPlain isBordered>
+          {credentialsSelected.map(c => (
+            <ListItem key={c.name}>{c.name}</ListItem>
+          ))}
+        </List>
+      </Modal>
       {connectionsSelected && (
         <ConnectionsModal source={connectionsSelected} connections={connectionsData} onClose={onCloseConnections} />
```

Comment:
**optional/future:**
this really should follow the same pattern of using a `isOpen` prop and allowing the use of
```
<ConnectionsModal
        isOpen={connectionsSelected !== undefined}
        source={connectionsSelected}
        connections={connectionsData}
        onClose={onCloseConnections}
      />
```
and
```
export interface ConnectionsModalProps {
  isOpen: boolean;
  source: SourceType | undefined;
  onClose: () => void;
  connections: {
    successful: ConnectionType[];
    failed: ConnectionType[];
    unreachable: ConnectionType[];
  };
}

export const ConnectionsModal: React.FC<ConnectionsModalProps> = ({ isOpen, source, onClose, connections }) => {
  const ...
```

**future:** 
for tracking ...  we have another `success... failed... and unreachable` we should really consider a centralized type for it

```
...: {
    successful: ConnectionType[];
    failed: ConnectionType[];
    unreachable: ConnectionType[];
  };
```